tasks/page/makeReadme.js

```diff
@@ -68,19 +65,25 @@ function addSpacing(string) {
 
 // replace blob file urls with raw file urls, they don't work outside of github
 function replaceBlobUrls(string) {
-  const imageUrl = /(https:\/\/github\.com\/.*?\/)(blob)(\/.*?\.)/gi
+  const blobUrl = /(https:\/\/github\.com\/.*?\/)(blob)(\/.*?\.)/gi
 
-  return string.replace(imageUrl, '$1raw$3')
+  return string.replace(blobUrl, '$1raw$3')
 }
 
 // add project path from root folder to images; required before being parsed by marky markdown
+function addProjectPath(imgTag, markdownPath, projectPath) {
+  const parts = markdownPath.split('/')
+  const fileName = parts[parts.length - 1]
+
+  return `${imgTag}/${projectPath}/${fileName}`
+}
+
 function correctMarkdownImagePaths(string, projectPath) {
-  const imagePath = /(!\[.*?]\()((?!https:\/\/).+?(\.png|\.jpg|\.gif|\.jpeg|\.bmp))/gi
+  const imagePath = /(!\[.*?]\()(.+?(\.png|\.jpg|\.gif|\.jpeg|\.bmp))/gi
 
-  return string.replace(imagePath, (_match, imgTag, path) => {
-    const parts = path.split('/')
-    const fileName = parts[parts.length - 1]
+  return string.replace(imagePath, (match, imgTag, path) => {
+    if (match.includes('/blob/')) return replaceBlobUrls(match)
 
-    return `${imgTag}/${projectPath}/${fileName}`
+    if (projectPath) return addProjectPath(imgTag, path, projectPath)
```

Comment:
Minor style stuff: 

- I never use if statements without brackets

Extra minor: 
- Commit messages always start with a capital

Reply:
More importantly: are these two really exclusive. I.e. "blob" or "project relative", never both?

Reply:
I don't believe so. Because we are looking at each match, it will either be a direct link (with a `blob`) or the image filename (we then add the path).

Reply:
Not sure why I stopped using a capital on the commit messages. Will fix those.